lib/src/realm_object.dart

```diff
@@ -369,6 +381,13 @@ class RealmException implements Exception {
   }
 }
 
+class UserCallbackException extends RealmException {
+  final Object userException;
```

Comment:
we are registering the exception instance and later we are assigning it to an object in the `UserCallbackError`. Should this property be an Exception instead? So at least the `message` can be accessed.

Reply:
My understanding is that in dart you can throw any object, which is why the catch clause receives an object as an argument. We're simply storing that and rethrowing it. E.g.

```dart
config.migrationCallback = (migration, oldVersion) {
	throw 5;
};
```

If we tried to cast that to an exception instance, we would fail.